view_conversations/conv-viewer-webapi/src/db_types.ts

```diff
@@ -0,0 +1,193 @@
+export interface OutputRow {
+  id: runtime_types.CommentId;
+  ancestor_id: runtime_types.CommentId;
+  authors: null | string[];
+  cleaned_content: null | string;
+  content: null | string;
+  conversation_id: runtime_types.ConversationId;
+  indentation: number;
+  page_id: number;
+  page_title: string;
+  parent_id: runtime_types.CommentId;
+  replyTo_id: runtime_types.CommentId;
+  rev_id: runtime_types.RevisionId;
+  timestamp: Date | null;
+  type: string;
+  user_id: number | null;
+  user_text: string | null;
+}
+
+
+interface SpannerFieldHandler<T> {
+  field_name : string;
+  fromSpannerResultField(row:spanner.ResultField) : T | null;
+}
+
+// TODO(ldixon): add additional constraints, e.g. regexp matching.
+class StringFieldHandler implements SpannerFieldHandler<string> {
+  constructor(public field_name: string) {
+    this.field_name = field_name;
+  }
+  public fromSpannerResultField(field:spanner.ResultField) : string | null {
+    if(!(field === null || typeof(field) === 'string')) {
+      console.log('Error: field:');
+      console.dir(field);
+      throw Error(`For ${this.field_name}: expected field:string, not: ${typeof(field)}`);
+    }
+    return field;
+  }
+}
+
+class BytesFieldHandler implements SpannerFieldHandler<string> {
+  constructor(public field_name: string) {
+    this.field_name = field_name;
+  }
+  public fromSpannerResultField(field:spanner.ResultField) : string | null {
+  if (field == null){
+    return null;
+  }
+  if(!(field instanceof Uint8Array)) {
+      console.log('Error: field:');
+      console.dir(field);
+      throw Error(`For ${this.field_name}: expected field: byte array, not: ${typeof(field)}`);
+  }
+  return ab2str(<Uint8Array>field);
+  }
+}
+
+class ArrayFieldHandler implements SpannerFieldHandler<string[]> {
+  constructor(public field_name: string) {
+    this.field_name = field_name;
+  }
+  public fromSpannerResultField(field:spanner.ResultField) : string[] | null {
+    if (field == null) {
+      return field;
+    }
+    if (!(field.constructor == Array)) {
+      console.log('Error: field:');
+      console.dir(field);
+      throw Error(`For ${this.field_name}: expected field: string array, not: ${typeof(field)}`);
+    }
+    let ret: string[] = []
+    for (let element of <string[]>field) {
+      ret.push(element)
+    }
+    return ret;
+  }
+}
+
+
+
+class IntFieldHandler implements SpannerFieldHandler<number> {
+  constructor(public field_name: string) {
+    this.field_name = field_name;
+  }
+  public fromSpannerResultField(field:spanner.ResultField) : number | null {
+    if(typeof(field) === 'string') {
+      throw Error(`For ${this.field_name}: expected field.value:string, but field:string`);
+    } else if (field instanceof Date) {
+      throw Error(`For ${this.field_name}: expected field.value:string, but field:Date`);
+    } else if (field === null) {
+      return null;
+    }
+    return parseInt((<{ value: string; }>field).value);
```

Comment:
It might be worth making {value:string} into its own type, since it's used in many places.